invenio_pidstore/cli.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Click command-line interface for PIDStore management."""
+
+from __future__ import absolute_import, print_function
+
+import click
+from flask_cli import with_appcontext
+from invenio_db import db
+
+
+def process_status(ctx, param, value):
+    """Return status value."""
+    from .models import PIDStatus
+
+    # Allow empty status
+    if value is None:
+        return None
+
+    if value not in PIDStatus.attributes:
+        raise click.BadParameter('Status needs to be one of {0}.'.format(
+            ', '.join(PIDStatus.attributes)
+        ))
+    return getattr(PIDStatus, value)
+
+
+#
+# PIDStore management commands
+#
+
+@click.group()
+def pid():
+    """PIDStore management commands."""
+
+
+@pid.command()
+@click.argument('pid_type')
+@click.argument('pid_value')
+@click.option('-s', '--status', default='NEW', callback=process_status)
+@click.option('-t', '--type', 'object_type', default=None)
+@click.option('-i', '--uuid', 'object_uuid', default=None)
+@with_appcontext
+def create(pid_type, pid_value, status, object_type, object_uuid):
+    """Create new persistent identifier."""
+    from .models import PersistentIdentifier
+
+    if bool(object_type) ^ bool(object_uuid):
+        raise click.BadParameter('Speficy both or any of --type and --uuid.')
+
+    new_pid = PersistentIdentifier.create(
+        pid_type,
+        pid_value,
+        status=status,
+        object_type=object_type,
+        object_uuid=object_uuid,
+    )
+    db.session.commit()
+    click.echo(new_pid.id)
```

Comment:
When a user creates a new PID by doing say `pid create recid 1234`, this command returns internal ID from the PID table.  Is this useful for CLI users? I guess not, because CLI users only use PID IDs and Object IDs in the other commands.  What about returning the created PID instead?  (Kinda like record CLI creation returns wanted UUID.)